### PR TITLE
Fix the broken kubemark-5 suite

### PR DESCRIPTION
**Pull request by ghost**
This is a rollback for PR  #25984 and #19242. 

The work here is typed by mhrgoog and directed by his overlords @lavalamp @mikedanese.

### CODE REVIEW

**Comment by mikedanese** [comment]
cc @wojtek-t @gmarek @andyzheng0831 @roberthbailey @mqliang

**Comment by andyzheng0831** [comment]
You guys have identified the root cause of the breakage? I will also revert the support from my PR #25927

**Comment by mhrgoog** [comment]
We are trying to root cause it now. We have a repro case.

**Comment by lavalamp** [comment]
We haven't decided if we should merge this or try for a fix yet. @mhrgoog and @mikedanese have a repro and are looking at it right now. Other things are blocking the queue at the moment so we don't have to decide right away.

**Comment by mhrgoog** [comment]
So mike and I poked at this and I think we found the issue. After sshing to
the master node on the kubemark master we saw that the controller was not
running.

We looked at the following files.

$ cat /var/log/controller.log
 I0520 22:21:43.539213    2933 plugins.go:71] No cloud provider specified.
 I0520 22:21:43.539397    2933 nodecontroller.go:157] Sending events to api
server.
 panic: runtime error: invalid memory address or nil pointer dereference
 [signal 0xb code=0x1 addr=0x0 pc=0x7f28ab]
 goroutine 1 [running]:
 panic(0x2303e40, 0xc82000e100)
         /usr/local/go/src/runtime/panic.go:481 +0x3e6

k8s.io/kubernetes/pkg/controller/node.(*NodeController).filterOutServiceRange(0xc820144000)
         /go/src/
k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/node/nodecontroller.go:342
+0x2b
 k8s.io/kubernetes/pkg/controller/node.(*NodeController).Run(0xc820144000,
0x2540be400)

...
$

The line in question is:

nodecontroller.go:342

if !nc.clusterCIDR.Contains(nc.serviceCIDR.IP.Mask(nc.clusterCIDR.Mask)) &&
!nc.serviceCIDR.Contains(nc.clusterCIDR.IP.Mask(nc.serviceCIDR.Mask)) {

Mike guessed that nc.serviceCIDR or nc.clusterCIDRwas a nil pointer.

Then we read controllers_flags

$ cat controllers_flags
 --v=2__
 --allocate-node-cidrs=true
 --cluster-cidr=10.240.0.0/13
 --terminated-pod-gc-threshold=100
$

Here he saw that SERVICE_CLUSTER_IP_RANGE is not set.

Mike figured out that this was due to a flag that is now required to be set
in order for the controller to be running was not set in the
cluster/kubemark/config-default.sh file. The flag is:
 SERVICE_CLUSTER_IP_RANGE.

Once we set the flag we re-ran test/kubemark/run-e2e-tests.sh and now it is
passing.

mrubin

File: cmd/kube-controller-manager/app/options/options.go [hunk 1]
```diff
@@ -142,8 +141,6 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
 	fs.StringVar(&s.ClusterName, "cluster-name", s.ClusterName, "The instance prefix for the cluster")
 	fs.StringVar(&s.ClusterCIDR, "cluster-cidr", s.ClusterCIDR, "CIDR Range for Pods in cluster.")
-	fs.StringVar(&s.ServiceCIDR, "service-cluster-ip-range", s.ServiceCIDR, "CIDR Range for Services in cluster.")
-	fs.Int32Var(&s.NodeCIDRMaskSize, "node-cidr-mask-size", s.NodeCIDRMaskSize, "Mask size for node cidr in cluster.")
```
Comment (eparis) [inline cmd/kube-controller-manager/app/options/options.go]: We can't just remove flags like this, can we? If anyone has them set the binary wouldn't launch, would it? Don't we need to mark them deprecated for a while?

Comment (mikedanese) [inline cmd/kube-controller-manager/app/options/options.go]: This flag was added in a PR that was merged this morning. This simply reverts that PR.

**Pull request closed without merging**